pkg/endpointslice/endpointslice.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2025 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpointslice
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// ListURLsFromUnstructured retrieves list of endpoint URLs from an unstructured object.
+// The URLs are expected to be present at `.status.endpoints[].url` path inside the object.
+func ListURLsFromUnstructured(endpointSlice unstructured.Unstructured) ([]string, error) {
```

Comment:
Can we generate test for this? Should be quick but I want to make sure we dont nil pointer exception here